src/client.ts

```diff
@@ -2419,6 +2419,29 @@ export class StreamChat<
         clonedMessage.user = { id: userId.id } as UserResponse<UserType>;
       }
     }
+
+    /**
+     * Server always expects mentioned_users to be array of string. We are adding extra check, just in case
+     * SDK missed this conversation.
+     */
+    if (
+      Array.isArray(clonedMessage.mentioned_users) &&
+      !isString(clonedMessage.mentioned_users[0])
```

Comment:
Can't we improve type detection here so that we don't need to keep following comments?

Reply:
Ideally according to our types, `clonedMessage` shouldn't be array of strings at all. But still we want to make sure that if its missed, we handle that ... so as to not break the API.

Reply:
So I think following type casting is necessary here. Not sure if we can avoid it without allowing array of user objects on `UpdatedMessage` type.

Reply:
Sounds good, let's apply suggestion from @mahboubii then I guess we can roll with it

Reply:
Ferhat, updated!!

Reply:
Thanks, do we need array type case in the final version ?